lib/rules/static-jsx.js

```diff
@@ -17,19 +21,12 @@ module.exports = {
       recommended: true,
     },
     fixable: 'code',
+    messages,
     schema: [],
   },
 
   create: function (context) {
-    const curlyBracePresenceInstance = curlyBracePresence.create({
-      ...context,
-      options: [
-        {
-          props: 'never',
-          children: 'ignore',
-        },
-      ],
-    })
```

Comment:
this spread cloning never worked actully
you can change `node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js` on current master version > run tests and you will see similar issue
```
static-jsx › invalid › (<FormattedMessage id={'greeting'} defaultMessage={`hello`} />)

    TypeError: context.getSourceCode is not a function
    Occurred while linting <input>:1

      52 |             attrNode.value.type === 'JSXExpressionContainer'
      53 |           ) {
    > 54 |             curlyBracePresenceInstance.JSXExpressionContainer(attrNode.value)
         |                                        ^
      55 |           }
      56 |         })
      57 |

      at lintUnnecessaryCurly (node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:234:26)
      at Object.JSXExpressionContainer (node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:371:11)
      at forEach (lib/rules/static-jsx.js:54:40)
          at Array.forEach (<anonymous>)
      at JSXOpeningElement (lib/rules/static-jsx.js:42:25)
```